### PR TITLE
Support MongoClient initialization with a URI

### PR DESCRIPTION
**Pull request by hakansakalli**
Mongo configuration : Mongodb URI connection [Doc](https://mongodb.github.io/node-mongodb-native/driver-articles/mongoclient.html#the-url-connection-format)

```
org.quartz.jobStore.mongoUri=mongodb://username:password@host_1:27017,host_2:27017,host_3:27017/database_name?replicaSet=name_replicaset
```

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1 +1 @@
-version=2.1.0-SNAPSHOT
+version=2.2.0-SNAPSHOT
```
Comment (michaelklishin) [inline gradle.properties]: Never alter version in a pull request. It's up to the maintainer to decide when to do that and what the new version should be.

File: src/main/java/com/novemberain/quartz/mongodb/db/MongoConnectorBuilder.java [hunk 2]
```diff
@@ -71,7 +71,7 @@ public MongoConnector build() throws SchedulerConfigException {
         }
 
         // Options below require database name
-        checkNotNull(dbName, "'Database name' parameter is required.");
+        //checkNotNull(dbName, "'Database name' parameter is required.");
```
Comment (michaelklishin) [inline src/main/java/com/novemberain/quartz/mongodb/db/MongoConnectorBuilder.java]: Why was this commented out?

File: src/main/java/com/novemberain/quartz/mongodb/db/MongoConnectorBuilder.java [hunk 3]
```diff
@@ -82,7 +82,8 @@ public MongoConnector build() throws SchedulerConfigException {
         if (uri != null) {
             // User passed URI.
             validateForUri();
-            return new InternalMongoConnector(writeConcern, uri, dbName);
+            MongoClientURI mongoUri=new MongoClientURI(uri);
```
Comment (michaelklishin) [inline src/main/java/com/novemberain/quartz/mongodb/db/MongoConnectorBuilder.java]: Please use spaces around the assignment operator.

File: gradle.properties (hunk 1 above)
Comment (hakansakalli) [inline gradle.properties]: Fixed.

File: src/main/java/com/novemberain/quartz/mongodb/db/MongoConnectorBuilder.java (hunk 2 above)
Comment (hakansakalli) [inline src/main/java/com/novemberain/quartz/mongodb/db/MongoConnectorBuilder.java]: Fixed. Moved after uri check.

File: src/main/java/com/novemberain/quartz/mongodb/db/MongoConnectorBuilder.java (hunk 3 above)
Comment (hakansakalli) [inline src/main/java/com/novemberain/quartz/mongodb/db/MongoConnectorBuilder.java]: Fixed.

**Comment by michaelklishin** [comment]
OK, after reviewing our tests it indeed appears that there's no obvious way to hook a basic test into without a separate `quartz.properties` file. So considering it good enough for now.

**Comment by michaelklishin** [comment]
I reverted it because CI failures are legit and can be reproduced locally.

**Pull request merged**